modules/ROOT/pages/syntax/expressions.adoc

```diff
@@ -1232,6 +1232,187 @@ RETURN COLLECT {
+* `LOCAL DATETIME`
+* `ZONED DATETIME`
+* `DURATION`
+* `POINT`
+
+For all available Cypher types, see the section on xref::values-and-types/property-structural-constructed.adoc#types-synonyms[types and their synonyms].
+
+.Query
+[source, cypher]
+----
+UNWIND [42, true, 'abc'] AS val
+RETURN val, val IS :: INTEGER AS isInteger
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| +val+ | +isInteger+
+| +42+ | +true+
+| +true+ | +false+
+| +'abc'+ | +false+
+2+d|Rows: 3
+|===
+
+[[type-predicate-expressions-not]]
+=== Type predicate expressions with NOT
+
+It is also possible to verify that a Cypher expression is not of a certain type, using the negated type predicate expression; `IS NOT ::`.
+
+.Query
+[source, cypher]
+----
+UNWIND [42, true, 'abc'] AS val
+RETURN val, val IS NOT :: STRING AS notString
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| +val+ | +notString+
+| +42+ | +true+
+| +true+ | +true+
+| +'abc'+ | +false+
+2+d|Rows: 3
+|===
+
+[[type-predicate-expressions-null]]
+=== Type predicate expressions for null
+All Cypher types are considered to include the `null` value.
+Therefore, `IS ::` will return `true` for all expressions evaluating to `null`, regardless of the type.
+
+.Query
+[source, cypher]
+----
+RETURN (toFloat(null)) IS :: BOOLEAN AS isBoolean
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| +isBoolean+
+| +true+
+1+d|Rows: 1
+|===
+
+Likewise, `IS NOT ::` will return `false` for all expressions evaluating to `null`, regardless of the type.
+
+.Query
+[source, cypher]
+----
+RETURN (null + 1) IS NOT :: DATE AS isNotDate
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| +isNotDate+
+| +false+
+1+d|Rows: 1
+|===
+
+[[type-predicate-expressions-properties]]
+=== Type predicate expression for properties
+Type predicate expressions can also be used to filter out nodes or relationships with properties of a certain type.
+
+A graph containing the following nodes is used for the example below:
+
+////
+[source, cypher, role=test-setup]
+----
+MATCH (n) DETACH DELETE n;
+CREATE (:Person {name: 'Alice', age:14}),
+(:Person {name:'Bob', age:'20'}),
+(:Person {name:'Charlie', age:21}),
+----
+////
+
+image:graph_expression_type_predicate.svg[]
+
+.Query
+[source, cypher]
+----
+MATCH (n:Person)
+WHERE n.age IS :: INTEGER AND n.age > 18
+RETURN n.name AS name, n.age AS age
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| +name+ | +age+
+| +Charlie+ | +21+
+2+d|Rows: 1
+|===
+
+
+[[type-predicate-expressions-number-sizes]]
+=== Type predicate expressions for numbers of different sizes
+For numerical values passed in as parameters, Cypher does not take the size of the number into account.
+Cypher will thus regard any exact numerical parameter as an `INTEGER`, regardless of if it is declared as e.g. an INT16 or an INT64 in the client programming language.
```

Comment:
This part is kinda hard to read: `regardless of if it is declared as e.g.`. It feels like it starts out one way and then halfway though switches to another?

Reply:
Cypher will therefore regard any exact numerical parameter as an `INTEGER` regardless of its declared size e.g. an INT16 or an INT32 passed through from a client programming language will both be treated as an `INTEGER`.